dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/strategy/DotAssetViewStrategy.java

```diff
@@ -0,0 +1,70 @@
+package com.dotmarketing.portlets.contentlet.transform.strategy;
+
+import static com.dotmarketing.portlets.contentlet.transform.strategy.TransformOptions.USE_ALIAS;
+import static com.dotmarketing.portlets.fileassets.business.FileAssetAPI.FILE_NAME_FIELD;
+import static com.dotmarketing.portlets.fileassets.business.FileAssetAPI.MIMETYPE_FIELD;
+import static com.dotmarketing.portlets.fileassets.business.FileAssetAPI.TITLE_FIELD;
+import static com.dotmarketing.util.UtilHTML.getIconClass;
+import static com.dotmarketing.util.UtilHTML.getStatusIcons;
+import static com.dotmarketing.util.UtilMethods.getFileExtension;
+
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.model.ResourceLink;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+import java.io.File;
+import java.util.Map;
+import java.util.Set;
+
+public class DotAssetViewStrategy extends WebAssetStrategy<Contentlet> {
+
+    private static final String ASSET = "asset";
+
+    DotAssetViewStrategy(final TransformToolbox toolBox) {
+        super(toolBox);
+    }
+
+    @Override
+    protected Map<String, Object> transform(final Contentlet dotAsset,
+            final Map<String, Object> map,
+            final Set<TransformOptions> options, final User user)
+            throws DotDataException, DotSecurityException {
+
+        String fileName = "unknown";
+        long fileSize = 0L;
+        try {
+            final File asset = dotAsset.getBinary(ASSET);
+            fileName = asset.getName();
+            fileSize = asset.length();
+        }catch (Exception e){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)